src/Crypto/Ed25519SignKey.cs

```diff
@@ -0,0 +1,83 @@
+using NSec.Cryptography;
+
+namespace Concordium.Sdk.Crypto;
+
+/// <summary>
+/// Represents a (secret) sign key of an ed25519 keypair.
+///
+/// Used for signing transactions.
+/// </summary>
+public record Ed25519SignKey : ISigner
+{
+    /// <summary>
+    /// The length of an ed25519 (secret) sign key in bytes.
+    /// </summary>
+    public const int SignKeyBytesLength = 32;
+
+    /// <summary>
+    /// Representation of the ed25519 (secret) sign key as a length-32 byte array.
+    /// </summary>
+    private readonly byte[] _value;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Ed25519SignKey"/> class.
+    /// </summary>
+    /// <param name="signKeyAsBytes">A length-32 byte array representing the ed25519 (secret) sign key.</param>
+    private Ed25519SignKey(byte[] signKeyAsBytes) => this._value = signKeyAsBytes;
+
+    /// <summary>
+    /// Gets the ed25519 (secret) sign key represented as a length-64 hex encoded string.
+    /// </summary>
+    public override string ToString() => Convert.ToHexString(this._value).ToLowerInvariant();
```

Comment:
is it returned as upper characters? if not then we could avoid one string allocation by calling `.ToLowerInvariant`

Reply:
Yea, `ToHexString` returns uppercase.